src/components/organisms/Nav.js

```diff
@@ -132,14 +184,42 @@ const Nav = () => {
             confirmText="Sign Up"
             cancelHandler={() => setModalState(modalStates.closed)}
           >
+            <label htmlFor="name">
+              Display Name
+              <input
+                type="text"
+                name="name"
+                value={isUsername}
+                onChange={(e) => { handleNameChange(e); }}
+                onBlur={(e) => { validation(e); }}
+                required
+              />
+            </label>
+              {nameError}
             <label htmlFor="email">
               Email
-              <input type="text" name="email" />
+              <input
+                type="text"
+                name="email"
+                required
+                value={isEmail}
+                onChange={(e) => { handleEmailChange(e); }}
+                onBlur={(e) => { validation(e); }}
+              />
             </label>
+              {emailError}
             <label htmlFor="password">
               Password
-              <input type="password" name="password" />
+              <input
+                type="password"
+                name="password"
+                value={isPassword}
```

Comment:
once you change the names of the states as I said above, make sure you refactor the references